### ISSUE TITLE
Add support for `Mint` and `Burn` receipts

### ISSUE THREAD

**Issue by deekerno** [FuelLabs/fuel-indexer#1205]
As I was going through the SDK upgrade to `fuels-rs v0.44` and `fuel-core v0.20`, I noticed that there are two new receipts that aren't covered: `Mint` and `Burn`; after the upgrade is finished, support should be added for these receipts.

**Comment by ra0x3** [comment]
- Blocked by https://github.com/FuelLabs/fuel-indexer/issues/1167
- Also https://github.com/FuelLabs/fuel-indexer/issues/1012

**Comment by deekerno** [comment]
This has been merged into `feature/beta-4-release`.
